search/search_indexes.py

```diff
@@ -33,14 +43,46 @@ def index_queryset(self, using=None):
 
     def prepare_text(self, obj):  # pylint: disable=no-self-use
         """Indexing of the primary content of a LearningResource."""
+        try:
+            # Strip XML tags from content before indexing.
+            tree = etree.fromstring(obj.content_xml)
+            content = etree.tostring(tree, encoding="utf-8", method="text")
+        except etree.XMLSyntaxError:
+            # For blank/invalid XML.
+            content = obj.content_xml
```

Comment:
@Ferdi What do you think? Is it common or valid for the content of a chunk of XML not to have valid XML inside of it?

@giocalitri: It may be that this would be rare or nonexistent in reality, since should normally be dealing with valid XML in OLX format, and that the only reason I encountered this was due to strings being passed during tests.